dateutil/test/property/test_isoparse_prop.py

```diff
@@ -0,0 +1,29 @@
+from hypothesis import given, assume
+from hypothesis import strategies as st
+
+from datetime import timedelta
+
+from dateutil import tz
+from dateutil.parser import isoparse
+
+import pytest
+
+# Strategies
+TIME_ZONE_STRATEGY = st.sampled_from([None, tz.tzutc()] +
+    [tz.gettz(zname) for zname in ('US/Eastern', 'US/Pacific',
+                                   'Australia/Sydney', 'Europe/London')])
+ASCII_STRATEGY = st.characters(max_codepoint=127)
+
+
+@pytest.mark.isoparser
+@given(dt=st.datetimes(timezones=TIME_ZONE_STRATEGY), sep=ASCII_STRATEGY)
+def test_auto(dt, sep):
```

Comment:
I like the round-trip test, seems like an ideal use case for hypothesis.  Only question is about the name `test_auto`.  Can that be clarified?  I'm guessing the isoparser marker is intended to pull some of the weight in describing the test?

Reply:
Yeah, I'll change it to `test_timespec_auto`, though some of the weight is pulled by the isoparser mark but more by the fact that it's a test in `test_isoparser_prop.py`